### PR TITLE
Properly escape TOML fields in `generate`

### PR DESCRIPTION
**Pull request by 00vareladavid**
Someone on slack hit a call to `hex` which was supposed to be deprecated.
Root cause of the issue was that `generate` did not properly escape the `authors` field.

I think just piping through `TOML.print` should do the trick.

Note I couldn't figure out how to keep the empty `deps` field when generating a project, so I just avoid printing it. Is it just there for aesthetic purposes? We don't care to preserve it when `rm`ing all dependencies...

### CODE REVIEW

**Comment by 00vareladavid** [comment]
bors try

File: ext/TOML/src/parser.jl [hunk 1]
```diff
@@ -585,7 +585,7 @@ function escape(p::Parser, st::Int, multiline::Bool)
                 NONE()
             end
         else
-            error(p, st, position(p), "unknown string escape: `$("\\x"*hex(ch,2))`")
+            error(p, st, position(p), "unknown string escape: `$("\\x"*string(UInt32(ch), base=16, pad=2))`")
```
Comment (StefanKarpinski) [inline ext/TOML/src/parser.jl]: Maybe assign to a variable on the previous line? This has gotten a bit verbose for splicing inline.

File: src/generate.jl [hunk 2]
```diff
@@ -46,20 +46,16 @@ function project(pkg::String, dir::String; preview::Bool)
...
-
-            [deps]
-            """
-        )
+        toml = Dict("authors" => authors,
+                    "name" => pkg,
+                    "uuid" => string(uuid),
+                    "version" => "0.1.0",
+                    )
+        TOML.print(io, toml, sorted=true, by=key -> (Types.project_key_order(key), key))
```
Comment (StefanKarpinski) [inline src/generate.jl]: Is this a related change?

Comment (00vareladavid) [reply src/generate.jl]: Yes, the reason the user hit the bug in the first place was that we were not properly escaping the authors field on `generate`. I think their git name had a backslash in it, which is not properly escaped if we just splice it in directly.

Comment (00vareladavid) [reply src/generate.jl]: I can make them separate commits if necessary

Comment (StefanKarpinski) [reply src/generate.jl]: No, that's cool, makes sense. I was just wondering about the connection.
